Loop/Managers/LoopDataManager.swift

```diff
@@ -1536,31 +1563,95 @@ extension LoopDataManager {
 
         // Calculate retrospective correction
+        let insulinSensitivity = settings.insulinSensitivitySchedule!.quantity(at: glucose.startDate)
+        let basalRate = settings.basalRateSchedule!.value(at: glucose.startDate)
+        let correctionRange = settings.glucoseTargetRangeSchedule!.quantityRange(at: glucose.startDate)
+
         retrospectiveGlucoseEffect = retrospectiveCorrection.computeEffect(
             startingAt: glucose,
             retrospectiveGlucoseDiscrepanciesSummed: retrospectiveGlucoseDiscrepanciesSummed,
             recencyInterval: LoopCoreConstants.inputDataRecencyInterval,
-            insulinSensitivitySchedule: settings.insulinSensitivitySchedule,
-            basalRateSchedule: settings.basalRateSchedule,
-            glucoseCorrectionRangeSchedule: settings.glucoseTargetRangeSchedule,
+            insulinSensitivity: insulinSensitivity,
+            basalRate: basalRate,
+            correctionRange: correctionRange,
             retrospectiveCorrectionGroupingInterval: LoopConstants.retrospectiveCorrectionGroupingInterval
         )
     }
 
     private func computeRetrospectiveGlucoseEffect(startingAt glucose: GlucoseValue, carbEffects: [GlucoseEffect]) -> [GlucoseEffect] {
+
+        let insulinSensitivity = settings.insulinSensitivitySchedule!.quantity(at: glucose.startDate)
+        let basalRate = settings.basalRateSchedule!.value(at: glucose.startDate)
+        let correctionRange = settings.glucoseTargetRangeSchedule!.quantityRange(at: glucose.startDate)
+
         let retrospectiveGlucoseDiscrepancies = insulinCounteractionEffects.subtracting(carbEffects, withUniformInterval: carbStore.delta)
         let retrospectiveGlucoseDiscrepanciesSummed = retrospectiveGlucoseDiscrepancies.combinedSums(of: LoopConstants.retrospectiveCorrectionGroupingInterval * retrospectiveCorrectionGroupingIntervalMultiplier)
         return retrospectiveCorrection.computeEffect(
             startingAt: glucose,
             retrospectiveGlucoseDiscrepanciesSummed: retrospectiveGlucoseDiscrepanciesSummed,
             recencyInterval: LoopCoreConstants.inputDataRecencyInterval,
-            insulinSensitivitySchedule: settings.insulinSensitivitySchedule,
-            basalRateSchedule: settings.basalRateSchedule,
-            glucoseCorrectionRangeSchedule: settings.glucoseTargetRangeSchedule,
+            insulinSensitivity: insulinSensitivity,
+            basalRate: basalRate,
+            correctionRange: correctionRange,
             retrospectiveCorrectionGroupingInterval: LoopConstants.retrospectiveCorrectionGroupingInterval
         )
     }
 
+    /// Generates a glucose prediction effect of suspending insulin delivery over duration of insulin action starting at current date
+    ///
+    /// - Throws: LoopError.configurationError
+    private func updateSuspendInsulinDeliveryEffect() throws {
```

Comment:
There are not. This effect isn't used in dosing. It's used in the forecast effects screen, to show the user the effects of a long suspend. I do think a test would be appropriate here, but it will need to be rewritten once we have a more functional algorithm, so I'd prefer to wait on it until after the algorithm updates are finished.